cmd/rook/ceph/operator.go

```diff
@@ -91,14 +111,132 @@ func startOperator(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		rook.TerminateFatal(err)
 	}
-
 	rookImage := rook.GetOperatorImage(context.Clientset, containerName)
 	serviceAccountName := rook.GetOperatorServiceAccount(context.Clientset)
 	op := operator.New(context, volumeAttachment, rookImage, serviceAccountName)
+	if isPresent, err := isSecretPresent(context); isPresent && err == nil {
+		err = initWebhook(context)
+		if err != nil {
+			logger.Errorf("an error has occurred while initializing webhook: %+v", err)
+		}
+	}
+	if err != nil {
+		logger.Errorf("error has occurred: %+v", err)
+	}
 	err = op.Run()
 	if err != nil {
 		rook.TerminateFatal(errors.Wrapf(err, "failed to run operator\n"))
 	}
 
 	return nil
 }
+
+func isSecretPresent(context *clusterd.Context) (bool, error) {
+
+	_, err := context.Clientset.CoreV1().Secrets(namespace).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		// If secret is not found. All good ! Proceed with rook without admission controllers
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, errors.Errorf("Error happened while retrieving the Secret: %+v", err)
```

Comment:
Use `Wrap` for errors that aren't logs.

Reply:
No capital letters in error log message if possible

Reply:
Fixed. Will just return an error